Resources/public/js/field/custom-entity-select.js

```diff
@@ -30,22 +30,30 @@ define(
                     fetcher,
                     SelectField.prototype.configure.apply(this, arguments)
                 ).then(function (items) {
-                    let config = this.config;
                     if (_.isEmpty(items)) {
                         this.config.readOnly = true;
                         this.config.choices = {
                             'NO OPTION': __('pim_custom_entity.import.csv.entity_name.no_reference_data')
                         };
                     } else {
                         let choices = {};
-                        let nameProp = this.config.choiceNameField;
-                        let valueProp = this.config.choiceValueField;
+                        const nameProp = this.config.choiceNameField;
+                        const valueProp = this.config.choiceValueField;
                         items.forEach(function (item) {
                             choices[item[nameProp]] = item[valueProp];
                         });
                         this.config.choices = choices;
                     }
                 }.bind(this));
+            },
+
+            /**
+             * {@inheritdoc}
+             */
+            getFieldValue: function (field) {
+                const value = $(field).val() || null;
```

Comment:
A bit hacky, but this allows to send `null` values in the request (instead of an empty string) after clearing the select2 component